packages/ckeditor5-engine/src/conversion/upcasthelpers.js

```diff
@@ -911,6 +923,40 @@ function prepareToAttributeConverter( config, shallow ) {
 	};
 }
 
+// TODO
+function prepareToAttributeConsumingConverter( config ) {
+	const matcher = new Matcher( config.view );
+
+	return ( evt, data, conversionApi ) => {
+		const match = matcher.match( data.viewItem );
+
+		// If there is no match, this callback should not do anything.
+		if ( !match ) {
+			return;
+		}
+
+		const modelValue = typeof config.model.value == 'function' ?
+			config.model.value( data.viewItem, conversionApi ) :
+			config.model.value;
+
+		// An attribute building function returned null so it was not converted so should not be consumed.
+		if ( modelValue === null ) {
+			return;
+		}
+
+		// We need to convert children if some higher priority converter did not convert them.
+		// After the element got consumed the default handler for converting children will not trigger.
+		if ( !data.modelRange ) {
+			// Convert children and set conversion result as a current data.
+			Object.assign( data, conversionApi.convertChildren( data.viewItem, data.modelCursor ) );
+		}
+
+		// Consume the element itself for element-to-attribute conversion so it won't get converted by any other converter
+		// since this element's whole purpose was to provide some attributes (for example <a href=""> or <span style="font-size: ...">).
+		conversionApi.consumable.consume( data.viewItem, { name: true } );
```

Comment:
I believe that _this_ is what prevents `<a>` from being generated by a downcast, because only `<a href="">` makes sense. Right?